service/matching/handler.go

```diff
@@ -40,16 +40,18 @@ var _ m.TChanMatchingService = (*Handler)(nil)
 type Handler struct {
 	taskPersistence persistence.TaskManager
 	engine          Engine
+	config          *Config
 	metricsClient   metrics.Client
 	startWG         sync.WaitGroup
 	service.Service
 }
 
 // NewHandler creates a thrift handler for the history service
-func NewHandler(taskPersistence persistence.TaskManager, sVice service.Service) (*Handler, []thrift.TChanServer) {
+func NewHandler(taskPersistence persistence.TaskManager, sVice service.Service, config *Config) (*Handler, []thrift.TChanServer) {
```

Comment:
nit: can we make the order of args to all handlers somewhat consistent.. i.e. can we make *Config always the first arg to NewHandler() for all services ?